src/record/api.rs

```diff
@@ -55,7 +56,113 @@ pub enum Row {
   Map(Vec<(Row, Row)>) // List of key-value pairs
 }
 
+// Macro to generate type-safe get_xxx methods e.g. get_bool, get_short
+macro_rules! row_primitive_accessor {
+  ($METHOD:ident, $VARIANT:ident, $TY:ty) => {
+    pub fn $METHOD(&self) -> Result<$TY, ParquetError> {
```

Comment:
nit: you can import `errors::Result` and replace `Result<$TY, ParquetError> ` with `Result<$TY>`